src/compas_fab/backends/kinematics/analytical_inverse_kinematics.py

```diff
@@ -0,0 +1,143 @@
+            a1, a2, a3, a4, a5, a6 = c.values()
+            try:
+                a1 = fit_within_bounds(a1, j1.limit.lower, j1.limit.upper)
+                a2 = fit_within_bounds(a2, j2.limit.lower, j2.limit.upper)
+                a3 = fit_within_bounds(a3, j3.limit.lower, j3.limit.upper)
+                a4 = fit_within_bounds(a4, j4.limit.lower, j4.limit.upper)
+                a5 = fit_within_bounds(a5, j5.limit.lower, j5.limit.upper)
+                a6 = fit_within_bounds(a6, j6.limit.lower, j6.limit.upper)
+                configurations[i].values = [a1, a2, a3, a4, a5, a6]
+            except AssertionError:
+                configurations[i] = None
+        return configurations
+
+
+class UR5_Analytical_IK(Analytical_InverseKinematics):
+
+    def _inverse_kinematics(self, frame):
+        from compas_fab.backends.kinematics.offset_wrist_kinematics import UR5
+        return UR5().inverse(frame)
+
+
+if __name__ == "__main__":
+    import time
+    import compas_fab
+    from compas_fab.robots import Tool
+    from compas.datastructures import Mesh
+    from compas.geometry import Point, Vector
+    from compas_fab.robots import RobotSemantics
+    from compas_fab.backends import PyBulletClient
+
+    class Client(PyBulletClient):
+        def inverse_kinematics(self, *args, **kwargs):
+            return UR5_Analytical_IK(self)(*args, **kwargs)
+
+    mesh = Mesh.from_stl(compas_fab.get('planning_scene/cone.stl'))
+
+    with Client() as client:
+        urdf_filename = compas_fab.get('universal_robot/ur_description/urdf/ur5.urdf')
+        srdf_filename = compas_fab.get('universal_robot/ur5_moveit_config/config/ur5.srdf')
+
+        # Load UR5
+        robot = client.load_robot(urdf_filename)
+        robot.semantics = RobotSemantics.from_srdf_file(srdf_filename, robot.model)
+
+        # Update disabled collisions
+        client.disabled_collisions = robot.semantics.disabled_collisions
+
+        # Attach tool and convert frames
+        robot.attach_tool(Tool(mesh, Frame((0.07, 0, 0), (0, 0, 1), (0, 1, 0)), name="light_pen"))
+        frames_WCF = [Frame(Point(0.407, 0.073, 0.320), Vector(0.922, 0.000, 0.388), Vector(0.113, 0.956, -0.269)),
+                      Frame(Point(0.404, 0.057, 0.324), Vector(0.919, 0.000, 0.394), Vector(0.090, 0.974, -0.210)),
+                      Frame(Point(0.390, 0.064, 0.315), Vector(0.891, 0.000, 0.454), Vector(0.116, 0.967, -0.228)),
+                      Frame(Point(0.388, 0.079, 0.309), Vector(0.881, 0.000, 0.473), Vector(0.149, 0.949, -0.278)),
+                      Frame(Point(0.376, 0.087, 0.299), Vector(0.850, 0.000, 0.528), Vector(0.184, 0.937, -0.296))]
+        frames_WCF_T0 = robot.attached_tool.from_tcf_to_t0cf(frames_WCF)
+
+        # Now check each frame
+        configurations_along_path = []
+
+        for frame in frames_WCF_T0:
+            # the following is not working because in robot we split joint_names and joint_values and do not support lists as outcomes
```

Comment:
I am not sure what you mean... That AnalyticIK returns configurations?

Reply:
That AnalyticIK returns a list of configurations, but other IKs return a configuration (not in a list).

Reply:
Ah, but having the possibility of retrieving not only one, but all 8 possible analytic solutions of a 6-axis robot is helpful for a lot of reasons, not only for the workshop.

Reply:
I completely agree.  I'm suggesting that there should be a discussion about whether other IKs should return lists with only one element or if the mixed return type is ok.

Reply:
Mixed return type is the least desirable option.
Lists of one element would work even if it feels a bit odd.
An alternative is to return a generator for all IK solvers, the analytical ones can yield all the possible ones (which are 8 only under some constrains but effectively can be more) and the numerical solvers can rely on that to yield new solutions until the client code stops iterating.

Reply:
The latter is obviously a breaking change.

Reply:
do we go for yield?

Reply:
yes

Reply:
and should we have `backends.interfaces.InverseKinematics.inverse_kinematics` yield a configuration instead of yielding joint positions and joint names